genesyscloud/knowledge_document_variation/data_source_genesyscloud_knowledge_document_variation.go

```diff
@@ -0,0 +1,38 @@
+package knowledgedocumentvariation
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"strings"
+	"terraform-provider-genesyscloud/genesyscloud/provider"
+	"terraform-provider-genesyscloud/genesyscloud/util"
+	"time"
+)
+
+func dataSourceKnowledgeDocumentVariationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := newVariationRequestProxy(sdkConfig)
+
+	name := d.Get("name").(string)
+	knowledgeBaseID := d.Get("knowledge_base_id").(string)
```

Comment:
Probably a good idea to use `buildVariationId()` and `splitVariationId()` functions to ensure consistency for handling.